File: React Native/Milk/backend/Milk.Data/Models/Product.cs
namespace Milk.Data.Models;

public class Product
{
    public string Id { get; set; } = Guid.NewGuid().ToString();

    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;

    public decimal Price { get; set; }
    public string Volume { get; set; } = string.Empty;

    public string ImageUrl { get; set; } = string.Empty;
    
    public string CategoryId { get; set; } = string.Empty;
    public Category Category { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: React Native/Milk/milk-mobile/types/product.ts
export interface Product {
    id: string;
    title: string;
    description: string;
    price: number;
    volume: string;
    imageUrl: string;
    categoryId: string;
    categoryName: string;
    categorySlug: string;
}

export interface PagedResponse<T> {
    page: number;
    pageSize: number;
    total: number;
    items: T[];
}